### ISSUE TITLE
Possibilities for adding IP source specific addresses

### ISSUE THREAD

**Issue by djsdevel** [tsduck/tsduck#29]
There are scenarios where I would like to look at transport streams with the same address and port and filter by the source address,  the DekTec tools, VLC and others allow this by using the convention 10.200.53.120@239.1.1.1:5000

Do you think the IP input for tsduck could be extended to support this some time in the future ?

**Comment by lelegard** [comment]
Having two streams with the same destination IP address and UDP port is a really strange idea. Nonetheless, the idea of filtering the stream by source address is very simple to implement and I will do it.

**Comment by lelegard** [comment]
Now, by default, multiple sources are detected and reported as warnings:
```
* Warning: ip: detected multiple sources for the same destination 239.255.1.1:59000 with potentially distinct streams
* Warning: ip: detected source: 192.168.56.101:59071
* Warning: ip: detected source: 192.168.56.102:47290
```

Two new options were added. The option `--source address[:port]` is used to filter packets based on a source address (and optional port). The option `--first-source` uses the sender `address:port` of the first received packet to establish the equivalent filter. In the latter case, you don't know in advance which source will be used but, at least, you will get a consistent unique stream, not a mixture of several sources.

The two options are, of course, mutually exclusive.

Fixed in commit ca3e3da62f3993f6d298113121a8363b74bd5fd8, version 3.8-525.
